### PR TITLE
Add interpolate option

### LINKED ISSUE #415: `interpolate` flag functionality is not implemented
I did `SECRET_KEY = env('DJANGO_SECRET_KEY')` and got the error:

```
django.core.exceptions.ImproperlyConfigured: Set the a&d3*0k#skh)!5o)$))zz9&%hr@-akid5zh--05ei790+wn0$r environment variable
```

This is becuase my `DJANGO_SECRET_KEY` starts with the dollar sign: 

```
DJANGO_SECRET_KEY="$a&d3*0k#skh)!5o)$))zz9&%hr@-akid5zh--05ei790+wn0$r"
```

There was a number of issues for this problem: #284, #60, #271.

Eventually, #333 somewhat solved the issue by allowing to escape the dollar sign. But this is not the best approach for generated values such as secret keys. The better option would be an ability to disable interpolation altogether. Ideally by default since the behaviour is unexpected.

Interestingly, the `interpolation` flag is described in the docs: https://django-environ.readthedocs.io/en/latest/tips.html?highlight=Proxied%20Values#proxy-value. BUT it's not implemented! [This commit ](https://github.com/joke2k/django-environ/commit/f8c8a1ff50109e1b6de99e3f6ad51b326a794392) introduced the doc. Apparently, the implementation (#145) was not merged and closed in favor of #333. I don't know why. #145 seems like a proper solution.

### PR DESCRIPTION
**Pull request by David-Wobrock**
The argument is already documented but not implemented yet.
Fixes #415

### CODE REVIEW

**Comment by David-Wobrock** [comment]
Hey @WTK 

Thanks for the comment, I think it would be worth creating another PR to fix the issue you are mentioning.
The hereby PR is adding the logic to deactivate the `interpolate` option, whereas, from what I'm understanding, you are mentioning that interpolation doesn't work properly.

**Comment by ThalusA** [comment]
@sergeyklay I see you are now the main contributor with push access to this repository, would you mind merging this simple PR ? It just makes interpolate works like it is supposed to be. It doesn't create new errors or anything like that.

**Comment by David-Wobrock** [comment]
Rebased to fix the conflict.

**Review by ThalusA — APPROVED** [review]

File: environ/environ.py
Comment (sergeyklay) [inline environ/environ.py:429]: line too long (97 > 79 characters)

**Review by sergeyklay — CHANGES_REQUESTED** [review]
@David-Wobrock I've been looking through your pull request and in principle, everything looks good. However, there seem to be a few builds that failed, notably the code style checks and linters. Could you please make the necessary adjustments? It would greatly assist in maintaining our codebase's quality and readability. Also, I would appreciate it if you could add an entry in the changelog about these changes as well as update corresponding article https://github.com/joke2k/django-environ/blob/main/docs/tips.rst#proxy-value. This would save me some time and ensure that we keep our documentation up to date.

**Comment by David-Wobrock** [comment]
Hi @sergeyklay 

Thanks for your input 👍 

> However, there seem to be a few builds that failed

I adapted the code ✔️ We should be good now (linting passes locally).
I forced-push, to keep one lean commit, but it seems that the PR requires new approval for running the CI 😕 

> add an entry in the changelog about these changes

Done ✔️ 

> update corresponding article https://github.com/joke2k/django-environ/blob/main/docs/tips.rst#proxy-value

From my understanding, everything is already documented in the tips section.
That was one of the major reasons of the PR: the logic was documented, but not implemented 😄

**Review by ThalusA — APPROVED** [review]

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thank you!

**Comment by sergeyklay** [comment]
@David-Wobrock Thank you for your contribution to the project. Unfortunately, after implementing your interpolation solution, [we've encountered issues from our users](https://github.com/joke2k/django-environ/issues/485). Due to this, we've decided to revert your changes and release a patch version.

This is not a reflection on the quality of your code or idea; it's just that it has caused some complications in this particular context.

**Pull request merged**
